### PR TITLE
fix missing dim out of range check for logcumsumexp_cuda with empty source tensor

### PR DESCRIPTION
**Pull request by yuguo68**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #78284

fix https://github.com/pytorch/pytorch/issues/73159

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -2335,6 +2335,15 @@ def logcumsumexp(a, axis):
                 'expected scalar_type Double but found Float'):
             torch.logcumsumexp(b, axis, out=inplace_out)
 
+        # Check dim out of range
```
Comment (ngimel) [inline test/test_torch.py:2338]: instead of adding error case here, can you please add ErrorInput to `logcumsumexp` OpInfo tests in common_methods_invocations. You'd need to add `error_inputs_func` to `logcumsumexp` entry, you can look at other examples to get an idea.

Comment (yuguo68) [reply test/test_torch.py:2338]: thanks, taking a look at `common_methods_invocations`

**Review by ngimel — APPROVED** [review]
lgtm, please land after CI passes
